$OpenBSD: patch-meson_build,v 1.2 2020/11/25 21:19:51 cwen Exp $

Hunk #1: ERROR: C shared or static library 'dl' not found
Hunk #2: fix the build on ld.bfd archs, where the -Bsymbolic-functions
         linker flag is not supported

Index: meson.build
--- meson.build.orig
+++ meson.build
@@ -125,18 +125,6 @@ config_h.set('ENABLE_DEBUG', enable_debug)
 # FIXME AC_USE_SYSTEM_EXTENSIONS also supported non-gnu systems
 config_h.set10('_GNU_SOURCE', true)
 
-# Check for functions
-
-libdl_dep = cc.find_library('dl')
-
-check_dl_functions_required = [
-  'dlsym',
-]
-
-foreach func: check_dl_functions_required
-  assert(cc.has_function(func, dependencies: libdl_dep), func + ' not found')
-endforeach
-
 # Compiler flags
 
 compiler_flags_common = [
@@ -226,8 +214,11 @@ linker_flags = [
 ]
 
 foreach flag: linker_flags
-  assert(cc.has_link_argument(flag), flag + ' is required but not supported')
-  add_project_link_arguments(flag, language: 'c',)
+  if cc.has_link_argument(flag)
+    add_project_link_arguments(flag, language: 'c',)
+  else
+    message(flag + ' is not supported')
+  endif
 endforeach
 
 # Dependencies
